action/protocol/rewarding/fund.go

```diff
@@ -146,3 +167,35 @@ func DepositGas(ctx context.Context, sm protocol.StateManager, amount *big.Int)
 	}
 	return rp.Deposit(ctx, sm, amount, iotextypes.TransactionLogType_GAS_FEE)
 }
+
+// DepositGasWithSGD deposits gas into the rewarding fund with Sharing of Gas-fee with DApps
+func DepositGasWithSGD(ctx context.Context, sm protocol.StateManager, sgdReceiver address.Address, totalAmount, sgdAmount *big.Int,
+) (*action.TransactionLog, error) {
+	if isZero(sgdAmount) {
```

Comment:
same with L150

Reply:
`isZero` is better, can handle the case `amount == nil`, will have another small PR to clean-up